### PR TITLE
Service invocation example using Java  and HTTP

### PR DESCRIPTION
**Pull request by pravinpushkar**
Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

# Description

Service invocation example using Java and HTTP client

## Issue reference

We strive to have all PRs being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: NA

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] The quickstart code compiles correctly
* [x] You've tested new builds of the quickstart if you changed quickstart code
* [x] You've updated the quickstart's README if necessary
* [x] If you have changed the steps for a quickstart be sure that you have updated the automated validation accordingly. All of our quickstarts have annotations that allow them to be executed automatically as code. For more information see [mechanical-markdown](https://github.com/dapr/mechanical-markdown#readme). For user guide with examples see [Examples](https://github.com/dapr/mechanical-markdown/tree/main/examples#mechanical-markdown-by-example).

### CODE REVIEW

File: service_invocation/java/http/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  - "== APP == Order passed: 2"
+  - "== APP == Order passed: 3"
+  - "== APP == Order passed: 4"
+  - "== APP == Order passed: 5"
+  - "== APP == Order passed: 6"
+  - "== APP == Order passed: 7"
+  - "== APP == Order passed: 8"
+  - "== APP == Order passed: 9"
+  - "== APP == Order passed: 10"
+expected_stderr_lines:
```
Comment (amulyavarote) [inline service_invocation/java/http/README.md:55]: Can have just a couple of lines for comparison

File: service_invocation/java/http/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  - "== APP == Order received: 2"
+  - "== APP == Order received: 3"
+  - "== APP == Order received: 4"
+  - "== APP == Order received: 5"
+  - "== APP == Order received: 6"
+  - "== APP == Order received: 7"
+  - "== APP == Order received: 8"
+  - "== APP == Order received: 9"
+  - "== APP == Order received: 10"
+expected_stderr_lines:
```
Comment (amulyavarote) [inline service_invocation/java/http/README.md:101]: Can have just a couple of lines for comparison.

**Review by amulyavarote — CHANGES_REQUESTED** [review]
Only the readme file changes are needed.

File: service_invocation/java/http/checkout/src/main/java/com/service/CheckoutServiceApplication.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+public class CheckoutServiceApplication {
+	private static final HttpClient httpClient = HttpClient.newBuilder()
+			.version(HttpClient.Version.HTTP_2)
+			.connectTimeout(Duration.ofSeconds(10))
+			.build();
+
+	private static final String DAPR_HTTP_PORT = System.getenv().getOrDefault("DAPR_HTTP_PORT", "3500");
+
+	public static void main(String[] args) throws InterruptedException, IOException {
+		String dapr_url = "http://localhost:"+DAPR_HTTP_PORT+"/neworder";
```
Comment (amulyavarote) [inline service_invocation/java/http/checkout/src/main/java/com/service/CheckoutServiceApplication.java:22]: Can you please change from "/neworder" to "/orders"

File: service_invocation/java/http/order-processor/src/main/java/com/service/controller/OrderProcessingServiceController.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class OrderProcessingServiceController {
+    @PostMapping(path = "/neworder", consumes = MediaType.ALL_VALUE)
+    public String processOrders(@RequestBody Order body) {
```
Comment (amulyavarote) [inline service_invocation/java/http/order-processor/src/main/java/com/service/controller/OrderProcessingServiceController.java:14]: Can you please change from "/neworder" to "/orders"

**Comment by amulyavarote** [comment]
Also, please rebase your branch with master.

**Review by paulyuk — APPROVED** [review]
LGTM!

**Pull request merged**